controller/authController.go

```diff
@@ -6,42 +6,125 @@ import (
 	"github.com/ayush/go-auth/config"
 	"log" 
 	"github.com/gin-gonic/gin"
+	"github.com/gorilla/sessions"
 )
 
+//The context "c" provides access to information about the current HTTP request.
 func SignUp(c *gin.Context) {
 	var user model.User
+
+	//The c.BindJSON -> Attempts to bind the JSON data from the request body to the user variable. 
+	//The c.BindJSON(&user) function automatically parses the JSON from the request body and populates the user struct.
 	if err := c.BindJSON(&user); err != nil {
 		log.Printf("Failed to bind JSON input for signup: %v", err)
-		c.JSON(http.StatusBadRequest, gin.H{"error": "cannot bind JSON input"})
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid input"})
 		return
 	}
 
 	err := user.SignUp(config.DB)
 	if err != nil {
-		log.Printf("Failed to sign up user: %v", err)
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "failed to sign up"})
+		// Sends back a JSON response
+		c.JSON(500, gin.H{"error": err})
```

Comment:
Why did you change this?

Reply:
I logging this error in User Model itself